site/src/pages/OrganizationSettingsPage/CustomRolesPage/CreateEditRolePage.tsx

```diff
@@ -48,8 +48,8 @@ export const CreateEditRolePage: FC = () => {
 	return (
 		<RequirePermission
 			isFeatureVisible={
-				organizationPermissions.assignOrgRoles ||
-				organizationPermissions.createOrgRoles
+				(organizationPermissions.updateOrgRoles && role !== undefined) ||
+				(organizationPermissions.createOrgRoles && role === undefined)
```

Comment:
```suggestion
				role ? organizationPermissions.updateOrgRoles : organizationPermissions.createOrgRoles
```

this feels a lot clearer to me: "if there's a role, check if we can update it, otherwise check if we can create"